src/which.js

```diff
@@ -47,20 +51,16 @@ function _which(options, cmd) {
 
       if (common.platform === 'win') {
         var baseAttempt = attempt;
-        attempt = baseAttempt + '.exe';
-        if (checkPath(attempt)) {
-          where = attempt;
-          return;
-        }
-        attempt = baseAttempt + '.bat';
-        if (checkPath(attempt)) {
-          where = attempt;
-          return;
-        }
-        attempt = baseAttempt + '.cmd';
-        if (checkPath(attempt)) {
-          where = attempt;
-          return;
+        var pathExtEnv = process.env.PATHEXT || XP_DEFAULT_PATHEXT;
+        var pathExtArray = splitPath(pathExtEnv);
+        var len = pathExtArray.length;
```

Comment:
Don't think array length caching makes any difference so you can just use `pathExtArray.length` in the `for` part without any performance penalty.